src/amo/pages/CategoryPage/index.js

```diff
@@ -0,0 +1,143 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import HeadLinks from 'amo/components/HeadLinks';
+import Page from 'amo/components/Page';
+import Search from 'amo/components/Search';
+import { ADDON_TYPE_EXTENSION } from 'amo/constants';
+import { withErrorHandler } from 'amo/errorHandler';
+import translate from 'amo/i18n/translate';
+import { fetchCategories } from 'amo/reducers/categories';
+import {
+  convertFiltersToQueryParams,
+  convertQueryParamsToFilters,
+  fixFiltersFromLocation,
+} from 'amo/searchUtils';
+import { apiAddonType } from 'amo/utils';
+import { getCategoryResultsPathname } from 'amo/utils/categories';
+import type { SearchFilters } from 'amo/api/search';
+import type { AppState } from 'amo/store';
+import type { ErrorHandlerType } from 'amo/types/errorHandler';
+import type { I18nType } from 'amo/types/i18n';
+import type { DispatchFunc } from 'amo/types/redux';
+import type { ReactRouterMatchType } from 'amo/types/router';
+
+type Props = {|
+  match: {|
+    ...ReactRouterMatchType,
+    params: {| categorySlug: string, visibleAddonType: string |},
+  |},
+|};
+
+type PropsFromState = {|
+  categoryName: string | null,
+  filters: SearchFilters,
+|};
+
+type InternalProps = {|
+  ...Props,
+  ...PropsFromState,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  i18n: I18nType,
+|};
+
+export class CategoryPageBase extends React.Component<InternalProps> {
+  getPageTitle(addonType: string, categoryName: string | null) {
+    const { i18n } = this.props;
+
+    // The only two possibilites for addonType are 'extension' and
+    // 'statictheme' because anything else would have thrown an
+    // exception in mapStateToProps via apiAddonType()
+    if (addonType === ADDON_TYPE_EXTENSION) {
+      return categoryName
+        ? i18n.sprintf(i18n.gettext('Extensions in %(categoryName)s'), {
+            categoryName,
+          })
+        : i18n.gettext('Extensions');
+    }
+
+    return categoryName
+      ? i18n.sprintf(i18n.gettext('Themes in %(categoryName)s'), {
+          categoryName,
+        })
+      : i18n.gettext('Themes');
+  }
+
+  constructor(props: InternalProps) {
+    super(props);
+
+    if (!this.props.categoryName) {
+      this.props.dispatch(
+        fetchCategories({ errorHandlerId: this.props.errorHandler.id }),
+      );
+    }
+  }
+
+  render() {
+    const { categoryName, filters, match } = this.props;
+    const { categorySlug, visibleAddonType } = match.params;
+    const addonType = apiAddonType(visibleAddonType);
```

Comment:
This could be injected in `mapStateToProps`.

Reply:
I started by doing that and it was creating an issue/complication, but I don't remember what it was. I can try again.